### PR TITLE
Fix scan capabilities query

### PR DESCRIPTION
**Pull request by yuumasato**
Check if oscap is available in remote machine before checking capabilities.

RHBZ#1368463

### CODE REVIEW

File: src/OscapScannerRemoteSsh.cpp [hunk 1]
```diff
@@ -107,10 +107,28 @@ void OscapScannerRemoteSsh::evaluate()
         return;
     }
 
-    emit infoMessage(QObject::tr("Querying capabilities on remote machine..."));
-
     {
         SshSyncProcess proc(mSshConnection, this);
+        emit infoMessage(QObject::tr("Checking if oscap is available on remote machine..."));
+
+        proc.setCommand(QString("hash"));
```
Comment (mpreisler) [inline src/OscapScannerRemoteSsh.cpp]: This relies on /bin/bash being the interpreter for the user. We may want to go with something more portable. Some people use zsh or even just sh.

File: src/OscapScannerRemoteSsh.cpp [hunk 2]
```diff
@@ -107,10 +107,28 @@ void OscapScannerRemoteSsh::evaluate()
...
+        proc.setArguments(QStringList(SCAP_WORKBENCH_REMOTE_OSCAP_PATH));
+        proc.setCancelRequestSource(&mCancelRequested);
+        proc.run();
+
+        if (proc.getExitCode() != 0)
+        {
+            emit errorMessage(
+                QObject::tr("Failed to locate oscap on remote machine. "
+                        "Please, check that openscap-scanner is installed on remote machine.")
```
Comment (mpreisler) [inline src/OscapScannerRemoteSsh.cpp]: Add "the": installed on the remote machine.

File: src/OscapScannerRemoteSsh.cpp (hunk 1 above)
Comment (yuumasato) [inline src/OscapScannerRemoteSsh.cpp]: Would `command -v` be a better option?

Comment (mpreisler) [inline src/OscapScannerRemoteSsh.cpp]: I think people mostly use `which` when they want to be compatible. It's in coreutils.

Comment (mpreisler) [inline src/OscapScannerRemoteSsh.cpp]: I will look into whether `command -v` is portable, I don't have a clue.

**Comment by mpreisler** [comment]
It seems to be portable - http://pubs.opengroup.org/onlinepubs/009696899/utilities/command.html

**Pull request merged**